### PR TITLE
test(rangeInput): add common test suite

### PR DESCRIPTION
**Pull request by sarahdayan**
This reworks the tests for the `rangeInput` widget to move them in the common test suite and reduce flavor-specific tests.

https://algolia.atlassian.net/browse/FX-2518

### CODE REVIEW

File: tests/common/widgets/range-input/options.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      });
+
+      ['min', 'max'].forEach((target) => {
+        expect(
+          document.querySelector(`.ais-RangeInput-input--${target}`)
+        ).toHaveAttribute('step', '0.01');
+      });
+    });
+
+    test('refines on submit', async () => {
```
Comment (Haroenv) [inline tests/common/widgets/range-input/options.ts:147]: not sure if worth testing, but we could add a test that only refines min or max, not both, to check it doesn't accidentally refine 0 or something.

**Review by Haroenv — APPROVED** [review]

File: tests/common/widgets/range-input/options.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  wait,
+} from '@instantsearch/testutils';
+import userEvent from '@testing-library/user-event';
+
+import type { RangeInputWidgetSetup } from '.';
+import type { TestOptions } from '../../common';
+
+function normalizeSnapshot(html: string) {
+  // React InstantSearch adds a noRefinement modifier on the list.
+  // @MAJOR: Remove this once React InstantSearch aligns with spec.
```
Comment (dhayab) [inline tests/common/widgets/range-input/options.ts:17]: Would it make sense to add the modifier in JS and Vue instead?

**Review by dhayab — APPROVED** [review]

File: tests/common/widgets/range-input/options.ts (hunk 2 above)
Comment (sarahdayan) [reply tests/common/widgets/range-input/options.ts:17]: We could, but to make it complete we would also need to introduce the `noRefinementRoot` className. Shouldn't we do this separately during our effort to unify?

Comment (dhayab) [reply tests/common/widgets/range-input/options.ts:17]: Yep I'm fine with this, thanks for looking!

File: tests/common/widgets/range-input/options.ts (hunk 1 above)
Comment (Haroenv) [reply tests/common/widgets/range-input/options.ts:147]: yes that's fine, although I meant to do that in a separate test

**Review by Haroenv — APPROVED** [review]

**Pull request merged**